src/com/xilinx/rapidwright/interchange/DeviceResourcesWriter.java

```diff
@@ -843,18 +925,43 @@ public static void writeAllWiresAndNodesToBuilder(Device device, DeviceResources
             wireBuilder.setType(wire.getIntentCode().ordinal());
         }
 
+        Map<Float, Integer> slowMaxDelayIndices = new HashMap<>();
+        // Make the zero index no delay
+        slowMaxDelayIndices.put(0f, 0);
+
         StructList.Builder<DeviceResources.Device.Node.Builder> nodeBuilders =
                 devBuilder.initNodes(allNodes.size());
         for (int i=0; i < allNodes.size(); i++) {
             DeviceResources.Device.Node.Builder nodeBuilder = nodeBuilders.get(i);
-            //Node node = allNodes.get(i);
             long nodeKey = allNodes.get(i);
             Node node = Node.getNode(device.getTile((int)(nodeKey >>> 32)), (int)(nodeKey & 0xffffffff));
             Wire[] wires = node.getAllWiresInNode();
             PrimitiveList.Int.Builder wBuilders = nodeBuilder.initWires(wires.length);
             for (int k=0; k < wires.length; k++) {
                 wBuilders.set(k, allWires.getIndex(makeKey(wires[k].getTile(), wires[k].getWireIndex())));
             }
+
+            if (delayEstimator != null && !node.isTied()) {
+                boolean incldueBase = false;
+                boolean includeDiscontinuity = true;
+                float delayPs = RouterHelper.computeNodeDelay(delayEstimator, node, incldueBase, includeDiscontinuity);
+                if (delayPs != 0) {
+                    int index = slowMaxDelayIndices.computeIfAbsent(delayPs, (v) -> slowMaxDelayIndices.size());
+                    nodeBuilder.setNodeTiming(index);
+                }
+            }
+        }
+
+        StructList.Builder<DeviceResources.Device.NodeTiming.Builder> nodeTimingsBuilder = devBuilder.initNodeTimings(slowMaxDelayIndices.size());
+        for (Map.Entry<Float,Integer> e : slowMaxDelayIndices.entrySet()) {
+            float slowMaxDelayPs = e.getKey();
+            int index = e.getValue();
+            DeviceResources.Device.NodeTiming.Builder timingBuilder = nodeTimingsBuilder.get(index);
+            // Delay represented as T = R * C where R = <delay in ps> and C = 1e-12 (1 ps)
+            DeviceResources.Device.CornerModelValues.Builder resBuilder = timingBuilder.initResistance().initSlow().initSlow();
+            resBuilder.initMax().setMax(slowMaxDelayPs);
+            DeviceResources.Device.CornerModelValues.Builder capBuilder = timingBuilder.initCapacitance().initSlow().initSlow();
+            capBuilder.initMax().setMax(1e-12f);
```

Comment:
Perhaps pull out `1e-12f` and other occurrences into a separate static variable with a note.